### DISCUSSION TITLE
Starlette Application Stops When ElasticAPM Server is Unreachable

### DISCUSSION THREAD

**Discussion by Impro02** [encode/starlette#2571] [Potential Issue] [answered]
### Environment:

Starlette version: 0.37.2
ElasticAPM version: 6.20.0
Uvicorn version: 0.27.1
Python version: 3.12.1
Operating System: Windows/Linux

### Issue Summary: 
When integrating ElasticAPM middleware with a Starlette application, the application ceases to function if the ElasticAPM server is unreachable. This occurs under various circumstances, such as an incorrect URL, proxy issues, or certificate verification failures.

### Steps to Reproduce:

- Set up a Starlette application with ElasticAPM middleware.
- Configure the ElasticAPM middleware to point to a non-existent APM server URL (to simulate server downtime).
- Start the Starlette application.
- Observe logging output indicating that the APM server cannot be reached.
- After approximately 30 seconds, the application stops.

### Expected Behavior: 
The Starlette application should continue to run and handle requests normally, even if the ElasticAPM server is unreachable. The ElasticAPM client should handle server unavailability gracefully, without impacting the application's availability.

### Actual Behavior: 
The Starlette application stops running within a minute after the ElasticAPM server becomes unreachable. This behavior is consistent across various scenarios where the APM server cannot be contacted.

### Additional Context: 
This issue poses a significant risk to application reliability, as any downtime or configuration issue with the ElasticAPM server directly affects the availability of all services using the Starlette framework with ElasticAPM middleware.

### Demo code to reproduce the issue:
```
from starlette.applications import Starlette
from starlette.routing import Route
from starlette.responses import JSONResponse

from elasticapm.contrib.starlette import (
    make_apm_client as _make_apm_client,
    ElasticAPM as _ElasticAPM,
)


apm = _make_apm_client(
    {
        "SERVICE_NAME": "demo",
        "ENVIRONMENT": "dev",
        "SERVER_URL": "https://demo-apm.com",
        "VERIFY_SERVER_CERT": True,
    }
)


async def status(request):
    return JSONResponse({"status": "ok"})


app = Starlette(
    routes=[
        Route("/status", status),
    ]
)

app.add_middleware(_ElasticAPM, client=apm)
```

```
uvicorn main:app --port 5000
```

**Comment by Kludex** [comment, marked as answer]
Doesn't look like a Starlette issue. Did you check with the Elastic team?

**Reply by Impro02** [reply]
This issue on elastic side:
https://github.com/elastic/apm-agent-python/issues/2031#issuecomment-2078783739
